web3/contracts/MirrorMarketplace.sol

```diff
@@ -0,0 +1,25 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+
+contract MirrorMarketplace {
+    uint listingId;
+
+    enum ListingStatus {
+        Listed,
+        Cancelled,
+        Sold
+    }
+
+    struct Listing {
+        IERC721 token;
+        address seller;
+        uint tokenId;
+        uint price;
+        uint timestamp;
+        ListingStatus status;
+    }
+
+    mapping (uint => Listing) private listings;
```

Comment:
whats the mapping for?

Reply:
so mapping is data structure in solidity. 
we need to have a list of listings to add and modify our listings.

solidity has array and mapping data structures to store lists